metrics.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 RedHat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	// MetricsEndpoint is the port to export metrics on
+	MetricsEndpoint = ":8080"
+)
+
+// StartMetrics register metrics and exposes them
+func StartMetrics(list []prometheus.Collector, fn recordmetrics) {
+	// Register metrics and start serving them on /metrics endpoint
+	RegisterMetrics(list)
+	//RecordMetrics
+	fn()
```

Comment:
What does `fn` do?

Reply:
fn() performs the user-defined operation on how to collect metrics. I have modified the naming to reflect the same.